### PR TITLE
Adds label summary dots to repo issues/PR cell

### LINKED ISSUE #511: Repo issues/PRs should show label summary (dots) on list view
Helps when browsing

### PR DESCRIPTION
**Pull request by joerocca**
labeldots

Closes #511

### CODE REVIEW

**Comment by rnystrom** [comment]
Super cool!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
Was afraid it was going to look too messy, but it looks awesome!

File: Classes/Repository/GQL+RepositoryIssueSummaryType.swift [hunk 1]
```diff
@@ -9,6 +9,10 @@
 import Foundation
 
 extension RepoIssuePagesQuery.Data.Repository.Issue.Node: RepositoryIssueSummaryType {
+    
+    var labelableFields: LabelableFields {
```
Comment (rnystrom) [inline Classes/Repository/GQL+RepositoryIssueSummaryType.swift]: reading "labelable" hurts my brain

File: Classes/Repository/RepositorySummaryCell.swift [hunk 2]
```diff
@@ -11,49 +11,62 @@ import SnapKit
...
-        contentView.addSubview(reasonImageView)
         reasonImageView.snp.makeConstraints { make in
             make.size.equalTo(Styles.Sizes.icon)
             make.centerY.equalToSuperview()
             make.left.equalTo(Styles.Sizes.columnSpacing)
         }
-
-        let inset = RepositorySummaryCell.titleInset
         
+        detailsStackView.axis = .vertical
```
Comment (rnystrom) [inline Classes/Repository/RepositorySummaryCell.swift]: My first adventure into stack views!

File: Classes/Views/DotListView.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        collectionView.reloadData()
```
Comment (rnystrom) [inline Classes/Views/DotListView.swift]: Can we remove this? It isn't necessary is it? Overkill.

File: Classes/Views/DotListView.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        collectionView.isUserInteractionEnabled = false
+        collectionView.register(IssueLabelDotCell.self, forCellWithReuseIdentifier: IssueLabelDotCell.reuse)
+        
+        super.init(frame: frame)
+        
+        collectionView.dataSource = self
+        addSubview(collectionView)
+        collectionView.snp.makeConstraints { (make) in
+            make.edges.equalTo(self)
+        }
```
Comment (rnystrom) [inline Classes/Views/DotListView.swift]: Could we move this to `layoutSubviews` and just do `collectionView.frame = bounds` instead? Cheaper, simpler.

File: Classes/Repository/RepositorySummaryCell.swift (hunk 2 above)
Comment (joerocca) [reply Classes/Repository/RepositorySummaryCell.swift]: Super useful for this!

File: Classes/Views/DotListView.swift (hunk 4 above)
Comment (joerocca) [reply Classes/Views/DotListView.swift]: Done!

File: Classes/Views/DotListView.swift (hunk 3 above)
Comment (joerocca) [reply Classes/Views/DotListView.swift]: This is actually here so the the max items are recalculated on orientation change.

dots

**Comment by rnystrom** [comment]
Couple merge conflicts on the project file if you don't mind!

**Comment by joerocca** [comment]
Good to go!

**Comment by rnystrom** [comment]
Lol sorry, conflicts again 😭

**Comment by BasThomas** [comment]
Not seeing any on sim, what's up with that?

**Pull request merged**